src/Compilers/Core/Desktop/AnalyzerFileReference.cs

```diff
@@ -208,13 +213,12 @@ internal void AddAnalyzers(ImmutableArray<DiagnosticAnalyzer>.Builder builder, s
                 // If there are no analyzers, don't load the assembly at all.
                 if (!analyzerTypeNameMap.ContainsKey(language))
                 {
-                    this.AnalyzerLoadFailed?.Invoke(this, new AnalyzerLoadFailureEventArgs(AnalyzerLoadFailureEventArgs.FailureErrorCode.NoAnalyzers, null, null));
```

Comment:
Interesting, this error has actually been quite helpful for me when I end up using an analyzer built against a different version Microsoft.CodeAnalysis nuget package then the toolset compiler used to build the assembly.

In absence of such a diagnostic, we will silently fail loading the analyzers from an actual analyzer assembly.

One other alternative is to still retain this diagnostic and add an additional check here to see if the assembly has at least one type whose base type name = "Microsoft.CodeAnalysis.DiagnosticAnalyzer"? We can probably also improve the error message to explicitly state that the analyzer assembly is build against a different toolset?